raiden/messages/encode.py

```diff
@@ -22,36 +22,22 @@
     SendSecretReveal,
     SendUnlock,
 )
-from raiden.utils.typing import MYPY_ANNOTATION
 
 
 def message_from_sendevent(send_event: SendMessageEvent) -> Message:
-    if type(send_event) == SendLockedTransfer:
-        assert isinstance(send_event, SendLockedTransfer), MYPY_ANNOTATION
-        return LockedTransfer.from_event(send_event)
-    elif type(send_event) == SendSecretReveal:
-        assert isinstance(send_event, SendSecretReveal), MYPY_ANNOTATION
-        return RevealSecret.from_event(send_event)
-    elif type(send_event) == SendUnlock:
-        assert isinstance(send_event, SendUnlock), MYPY_ANNOTATION
-        return Unlock.from_event(send_event)
-    elif type(send_event) == SendSecretRequest:
-        assert isinstance(send_event, SendSecretRequest), MYPY_ANNOTATION
-        return SecretRequest.from_event(send_event)
-    elif type(send_event) == SendLockExpired:
-        assert isinstance(send_event, SendLockExpired), MYPY_ANNOTATION
-        return LockExpired.from_event(send_event)
-    elif type(send_event) == SendWithdrawRequest:
-        assert isinstance(send_event, SendWithdrawRequest), MYPY_ANNOTATION
-        return WithdrawRequest.from_event(send_event)
-    elif type(send_event) == SendWithdrawConfirmation:
-        assert isinstance(send_event, SendWithdrawConfirmation), MYPY_ANNOTATION
-        return WithdrawConfirmation.from_event(send_event)
-    elif type(send_event) == SendWithdrawExpired:
-        assert isinstance(send_event, SendWithdrawExpired), MYPY_ANNOTATION
-        return WithdrawExpired.from_event(send_event)
-    elif type(send_event) == SendProcessed:
-        assert isinstance(send_event, SendProcessed), MYPY_ANNOTATION
-        return Processed.from_event(send_event)
-    else:
+    event_map = {
+        SendLockExpired: LockExpired,
+        SendLockedTransfer: LockedTransfer,
+        SendProcessed: Processed,
+        SendSecretRequest: SecretRequest,
+        SendSecretReveal: RevealSecret,
+        SendUnlock: Unlock,
+        SendWithdrawConfirmation: WithdrawConfirmation,
+        SendWithdrawExpired: WithdrawExpired,
+        SendWithdrawRequest: WithdrawRequest,
+    }
+    t_event = type(send_event)
+    EventClass = event_map.get(t_event)
+    if EventClass is None:
         raise ValueError(f"Unknown event type {send_event}")
+    return EventClass.from_event(send_event)  # type: ignore
```

Comment:
I'd move the event map out of the function since its content is static and we don't need to create a dict every time this function is called (please take care to make it private if you do so).

Also, I'm not sure what's the value of raising `ValueError` here. Do we actually expect it somewhere? If not, I'd suggest just doing a regular lookup in the dict and letting Python raise the usual `KeyError`.

Reply:
I agree on the part of moving the map out of the function, but raising the `ValueError` gives us a clearer error message than the `KeyError`, so I'd rather keep it that way.